workers/unity/Packages/com.improbable.gdk.core/Subscriptions/EntityGameObjectLinker.cs

```diff
@@ -51,19 +52,17 @@ public EntityGameObjectLinker(World world)
         public void LinkGameObjectToSpatialOSEntity(EntityId entityId, GameObject gameObject,
             params Type[] componentTypesToAdd)
         {
-            if (!workerSystem.TryGetEntity(entityId, out var entity))
+            var linkedComponent = gameObject.GetComponent<LinkedEntityComponent>();
+            if (linkedComponent == null)
             {
-                throw new ArgumentException("Entity not in view");
+                linkedComponent = gameObject.AddComponent<LinkedEntityComponent>();
             }
-
-            if (gameObjectToComponentsAdded.TryGetValue(gameObject, out var added))
+            else if (linkedComponent.IsValid)
```

Comment:
What does it mean if the `linkedComponent` is not valid?

Reply:
Means it isn't linked to a gameobject.

Reply:
It could do with a better name.